src/edu/csus/ecs/pc2/core/report/ResultsCompareReport.java

```diff
@@ -59,6 +59,49 @@ public ResultsCompareReport(IInternalContest contest, IInternalController contro
         this(contest, controller, primaryCCSResultsDir, pc2ResultsDir, fullDetails, null, null);
     }
     
+    /**
+     * Validate input for report, check parameters, check for missing dirs and invalid input files.
+     */
+    public void validateInput() {
+
+        String sourceDir = pc2ResultsDir;
+        String targetDir = primaryCCSResultsDir;
+
+        if (targetDir == null) {
+            throw new RuntimeException("Primary CCS directory is not assigned (is null)");
+        }
+
+        if (sourceDir == null) {
+            throw new RuntimeException("PC2 directory is not assigned (is null)");
+        }
+
+        if (!Utilities.isDirThere(targetDir)) {
+            throw new RuntimeException("Primary CCS results dir: " + primaryCCSResultsDir + " does not exist");
+        }
+
+        if (!Utilities.isDirThere(sourceDir)) {
+            throw new RuntimeException("PC2 results dir: " + sourceDir + " does not exist");
+        }
+
+        if (primaryCCSResultsDir == null || !(new File(primaryCCSResultsDir).isDirectory())) {
+            throw new RuntimeException("Primary CCS Results directory not defined or not a directory");
+        }
+
+        if (pc2ResultsDir == null || !(new File(pc2ResultsDir).isDirectory())) {
```

Comment:
This check is redundant; the exact same conditions have already been checked above and if they were true an exception was thrown, so the code inside this `if` can never be executed.